vidpub/__main__.py

```diff
@@ -123,6 +123,13 @@ def parse_args(argv):
     )
     return parser.parse_args(argv)
 
+def media_batch_reader(file_path, chuncksize=64 * (1 << 20)):
+    print(f"Reading Vedio from:\n\t{file_path}")
+    blocks = b""
+    with open(file_path, "rb") as f:
+        for block in tqdm.tqdm(iter(functools.partial(f.read, chuncksize), b''), total=int(file_path.stat().st_size/chuncksize)):
+            blocks += block
+    return blocks
```

Comment:
Actually, probably better to use `io.BytesIO` here

```suggestion
    out = io.BytesIO()
    reader = iter(functools.partial(f.read, chuncksize)
    total = file_path.stat().st_size // chuncksize
    with open(file_path, "rb") as f:
        for block in tqdm.tqdm(reader, b''), total=total):
            out.write(block)
    return out.getvalue()
```

Alternatively…

```python
def media_batch_reader(file_path, chuncksize=64 * (1 << 20)):
    reader = iter(functools.partial(f.read, chuncksize)
    total = file_path.stat().st_size // chuncksize
    with open(file_path, "rb") as f:
        for block in tqdm.tqdm(reader, b''), total=total):
            yield block

...

media = apiclient.http.MediaInMemoryUpload(
    b"".join(media_batch_reader(vid_path)), resumable=True
)
```

Reply:
Yeah. that's right. Thanks for the suggest. I've updated it to use the first version. Please have a look. Thanks!